service/worker/indexer/esProcessor.go

```diff
@@ -64,6 +65,11 @@ type (
 		logger        log.Logger
 		metricsClient metrics.Client
 	}
+
+	kafkaMessageWithMetrics struct { // value of esProcessorImpl.mapToKafkaMsg
```

Comment:
you can also make this struct implement messaging.Message and make the implemented ack / nack call stopwatch.stop

Reply:
updated, I think your suggestion is cleaner